lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -307,10 +307,23 @@ defmodule Archethic.Mining.PendingTransactionValidation do
          type: :nft,
          data: %TransactionData{content: content}
        }) do
-    if Regex.match?(~r/(?<=initial supply:).*\d/mi, content) do
+    schema =
+      :archethic
+      |> Application.app_dir("priv/json-schemas/token-core.json")
+      |> File.read!()
+      |> Jason.decode!()
+      |> ExJsonSchema.Schema.resolve()
+
+    with {:ok, json_token} <- Jason.decode(content),
+         :ok <- ExJsonSchema.Validator.validate(schema, json_token) do
```

Comment:
You should add a verification to attest that if the properties list have more than 1 item, the supply must be the same than the properties list length